projects/online/online/utils/searcher.py

```diff
@@ -0,0 +1,181 @@
+import json
+import logging
+import os
+import time
+from dataclasses import asdict, dataclass
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import List, Optional
+
+import numpy as np
+from gwpy.time import tconvert
+from ledger.events import EventSet
+from online.utils.dataloading import get_prefix
+
+SECONDS_PER_YEAR = 31556952  # 60 * 60 * 24 * 365.2425
+
+
+def gps_from_timestamp(timestamp: float):
+    return float(tconvert(datetime.fromtimestamp(timestamp, tz=timezone.utc)))
+
+
+@dataclass
+class Event:
+    gpstime: float
+    detection_statistic: float
+    far: float
+    ifos: List[str]
+    datadir: Path
+    ifo_suffix: Optional[str] = None
+
+    def get_frame_write_time(self):
+        """
+        Get the time the frame corresponding
+        to this event was written to disk
+        """
+        t_write = 0
+        if self.ifo_suffix is not None:
+            ifo_dir = "_".join([self.ifos[0], self.ifo_suffix])
+        else:
+            ifo_dir = self.ifos[0]
+        prefix, length, _ = get_prefix(self.datadir / ifo_dir)
+        middle = "_".join(prefix.split("_")[1:])
+        for ifo in self.ifos:
+            if self.ifo_suffix is not None:
+                ifo_dir = "_".join([ifo, self.ifo_suffix])
+            else:
+                ifo_dir = ifo
+            prefix = f"{ifo[0]}-{ifo}_{middle}"
+            fname = (
+                self.datadir
+                / ifo_dir
+                / f"{prefix}-{int(self.gpstime)}-{length}.gwf"
+            )
+            t_write = max(t_write, os.path.getmtime(fname))
+
+        return gps_from_timestamp(t_write)
+
+    def __str__(self):
+        return (
+            "Event("
+            f"gpstime={self.gpstime:0.3f}, "
+            f"detection_statistic={self.detection_statistic:0.2f}, "
+            f"far={self.far:0.3e} Hz"
+            ")"
+        )
+
+    @property
+    def filename(self):
+        return f"event_{int(self.gpstime)}.json"
+
+    def write(self, directory: Path):
+        """
+        Write event information to a local directory
+        """
+        path = directory / f"event_{int(self.gpstime)}"
+        path.mkdir(exist_ok=True, parents=True)
+        filename = path / f"event-{int(self.gpstime)}.json"
+
+        event = asdict(self)
+        filecontents = str(event)
+        filecontents = json.loads(filecontents.replace("'", '"'))
+        with open(filename, "w") as f:
+            json.dump(filecontents, f)
+
+        return path
+
+
+class Searcher:
```

Comment:
We might want to rename this in light of the new function in `main`, but that's a second or third order correction